tests/csapi/direct_message_create_test.go

```diff
@@ -0,0 +1,56 @@
+package csapi_tests
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/tidwall/gjson"
+)
+
+func TestDirectMessageCreate(t *testing.T) {
```

Comment:
Not a good test name. I had to read https://github.com/matrix-org/synapse/issues/9768 to figure out what the regression test here is for. Please at the very least mention duplicate membership event _somewhere_.